src/main/java/org/runejs/client/Main.java

```diff
@@ -50,6 +54,17 @@
 import java.net.Socket;
 
 public class Main extends GameShell {
+
+    /**
+     * The codec currently in use to encode and decode packets.
+     */
+    public static final MessagePacketCodec packetCodec = new RuneJS435PacketCodec();
```

Comment:
Would it be an idea to place a `CodecLoader` in between to manage which codec(s) get loaded instead of importing them all in the `Main` class? Just thinking about extending the possibility of loading multiple codecs at once for multi-rev caches instead of having to Frankenstein a new codec for every possibility.

Just a thought.

Reply:
I agree, but I think we can probably leave this for now and revisit. Because:

- only a single `MessagePacketCodec` will be in use at one time, in the situation you describe I think we will probably create a new `MessagePacketCodec` for the combination we desire, and it will be composed of encoders/decoders used in other revisions.
- there are only 3 usages of this and it's the "plumbing" of the system - I don't expect this number to change. Nothing else should be depending on this `packetCodec` value.
    - Likewise for `handlerRegistry` but that only has 1 usage 😍

Reply:
So if someone built a 317 codec for instance they'd just change this to

```
public static final MessagePacketCodec packetCodec = new RS317PacketCodec();
```

Reply:
Fair enough. Can you create a ticket so we don't forget about it?

Reply:
ticket made 💪 https://github.com/runejs/refactored-client-435/issues/93